capplets/common/mate-theme-apply.c

```diff
@@ -68,6 +85,11 @@ mate_meta_theme_set (MateThemeMetaInfo *meta_theme_info)
   if (compare (old_key, meta_theme_info->gtk_theme_name))
     {
       g_settings_set_string (interface_settings, GTK_THEME_KEY, meta_theme_info->gtk_theme_name);
+
+      if ((wayland == TRUE) && (interface_gnome_settings))
```

Comment:
again, do we really need to check both here?

Reply:
Changed that to just checking for `wayland = TRUE` in the latest force-push